datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/ci.py

```diff
@@ -49,7 +49,13 @@
 
 
 def sort_jobs(jobs):
-    return sorted(jobs, key=lambda job: (not job['checkName'].startswith('datadog_checks_'), job['checkName']))
+    return sorted(
+        jobs,
+        key=lambda job: (
+            not job.get('checkName', '').startswith('datadog_checks_'),
+            job.get('checkName') or list(job.values())[0][0]['checkName'],
+        ),
+    )
```

Comment:
With the other suggestion this would become:

```suggestion
    return sorted(
        jobs,
        key=lambda job: (
            not get_check_name_from_job(job).startswith('datadog_checks_'),
            get_check_name_from_job(job),
        ),
    )
```